Lib/test/pickletester.py

```diff
@@ -2994,6 +2995,70 @@ def __reduce__(self):
 class BBB(object):
     pass
 
+
+class AbstractHookTests(unittest.TestCase):
+    def test_pickler_hook(self):
+        # test the CPickler ability to register custom, user-defined reduction
+        # callbacks to pickle functions and classes.
+
+        def custom_reduction_callback(pickler, obj):
+            obj_name = obj.__name__
+
+            if obj_name == 'f':
+                # asking the pickler to save f as 5
+                return int, (5, )
+
+            if obj_name == 'MyClass':
+                return str, ('some str',)
+
+            elif obj_name == 'log':
```

Comment:
This needn't be tested explicitly, since you have a fallback `return NotImplemented` at the end.